src/compas/files/base_reader.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import
+from __future__ import division
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+class BaseReader(object):
+    """Base class containing file reading functions for file extension specific readers
+
+    Attributes
+    ----------
+    location : str or pathlib object
+        Path or URL to the file
+    """
+    def __init__(self, location):
+        self.is_url = False
+        self.location = self.check_location(location)
+
+    def read(self, mode='text'):
```

Comment:
i find it confusing that the `read` method does something else than the default method of a file object

Reply:
if this is literally read line-by-line, then why not call this `read_line` and keep the rest of the names as `read` ?

Reply:
Renamed it read_line_or_chunk but that feels quite clunky too. iterate_over_file?

Reply:
`iterlines`?

Reply:
Instead of having one method that does two things and receives a `mode` to `if` on it, why not have two methods properly named: `read_line()` and `read_bytes()` and ditch the mode parameter?